components/menu/menu.js

```diff
@@ -76,13 +76,31 @@ class Menu extends HierarchicalViewMixin(LitElement) {
 	}
 
 	focus() {
-		if (this._items[0].role === 'menuitemradio') {
+		if (this.getMenuType() === 'menu-radio') {
 			this._focusSelected();
 		} else {
 			this._focusFirst();
 		}
 	}
 
+	getMenuType() {
+		if (this._items.length > 0) {
+			switch (this._items[0].role) {
+				case 'menuitemradio':
+					return 'menu-radio';
+
+				case 'menuitemcheckbox':
+					return 'menu-checkbox';
+
+				case 'menuitem':
+				default:
+					return 'menu';
+			}
+		} else {
+			return 'menu';
+		}
```

Comment:
Super nit: I like using `if`-statement to bail out quickly and then reduce the nesting of the rest of my blocks. In cases like this it also results in less code. For example:

```
if (this._items.length === 0) {
    return 'menu';
}
switch (this._items[0].role) {
	case 'menuitemradio':
		return 'menu-radio';
	case 'menuitemcheckbox':
		return 'menu-checkbox';
	case 'menuitem':
	default:
		return 'menu';
}
```